alphadia/transferlearning/train.py

```diff
@@ -300,6 +298,141 @@ def _order_intensities(
+        property_name: str,
+    ):
+        """
+        Accumulate the training metrics (training loss and learning rate) for the given property.
+
+        Parameters
+        ----------
+        metric_accumulator : MetricManager
+            The metric manager object.
+        epoch : int
+            The current epoch number.
+        epoch_loss : float
+            The training loss value of the current epoch.
+        current_lr : float
+            The current learning rate.
+        property_name : str
+            The property name to accumulate the metrics for.
+        """
+        loss_name = "ce_loss" if property_name == "charge" else "l1_loss"
+        metric_accumulator.accumulate_metrics(
+            epoch,
+            metric=epoch_loss,
+            metric_name=loss_name,
+            data_split="train",
+            property_name=property_name,
+        )
+        metric_accumulator.accumulate_metrics(
+            epoch,
+            metric=current_lr,
+            metric_name="lr",
+            data_split="train",
+            property_name=property_name,
+        )
+
+    def _evaluate_metrics(
+        self,
+        test_input: dict,
+        metric_accumulator: MetricManager,
+        epoch: int,
+        data_split: str,
+        property_name: str,
+        epoch_loss: float,
+        current_lr: float,
+    ) -> bool:
+        """
+        Evaluate the model using the test_input, accumulate the metrics and return the continue_training flag based on the early stopping criteria.
+
+        Parameters
+        ----------
+        test_input : dict
+            The input data for calculating the metrics.
+        metric_accumulator : MetricManager
+            The metric manager object.
+        epoch : int
+            The current epoch number.
+        data_split : str
+            The dataset label to test on e.g. "validation", "train"
+        property_name : str
+            The property name to accumulate the metrics for.
+        epoch_loss : float
+            The training loss value of the current epoch.
+        current_lr : float
+            The current learning rate.
+
+        Returns
+        -------
+        bool
+            Whether to continue training or not based on the early stopping criteria applied on the metrics.
+        """
+        continue_training = True
+        val_metrics = metric_accumulator.calculate_test_metric(
+            test_input, epoch, data_split=data_split, property_name=property_name
+        )
+        if epoch != -1 and data_split == "validation":
+            loss_name = "ce_loss" if property_name == "charge" else "l1_loss"
+            self._accumulate_training_metrics(
+                metric_accumulator, epoch, epoch_loss, current_lr, property_name
+            )
+            val_loss = val_metrics[val_metrics["metric_name"] == loss_name][
+                "value"
+            ].values[0]
+            continue_training = self.early_stopping.step(val_loss)
```

Comment:
(nit) if you `return self.early_stopping.step(val_loss)` here, you can drop the `continue_training` variables and the `else`